moda-contracts/src/Releases.sol

```diff
@@ -0,0 +1,204 @@
+    error CallerIsNotReleaseAdmin();
+    error InvalidTokenId();
+    error CallerDoesNotHaveTrackAccess();
+
+    /**
+     * @dev Constructor
+     * @param isInitializerDisabled - Whether the initializer is disabled or not
+     */
+    constructor(bool isInitializerDisabled) {
+        if (isInitializerDisabled) {
+            _disableInitializers();
+        }
+    }
+
+    // External Functions
+
+    /**
+     * @inheritdoc IReleases
+     */
+    function initialize(
+        address admin,
+        address[] calldata releaseAdmins,
+        string calldata name_,
+        string calldata symbol_,
+        address catalog,
+        address splitsFactory
+    ) external initializer {
+        __ERC1155_init("");
+        if (admin == address(0)) revert CannotBeZeroAddress();
+        if (bytes(name_).length == 0) revert FieldCannotBeEmpty("name");
+        if (bytes(symbol_).length == 0) revert FieldCannotBeEmpty("symbol");
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+        for (uint8 i = 0; i < releaseAdmins.length; i++) {
+            _grantRole(RELEASE_ADMIN_ROLE, releaseAdmins[i]);
+        }
+        name = name_;
+        symbol = symbol_;
+        _catalog = catalog;
+        _splitsFactory = splitsFactory;
+    }
+
+    /**
+     * @inheritdoc IReleases
+     */
+    function create(
+        address receiver,
+        uint96 royaltyAmount,
+        address beneficiary,
+        /// will be an array of addresses to pass to splits factory
+        string calldata uri_,
+        uint256 amount,
+        string[] calldata trackIds
+    ) external nonReentrant {
+        if (_isOpen) {
+            _requireCallerHasTrackAccess(trackIds);
+        } else if (!hasRole(RELEASE_ADMIN_ROLE, msg.sender)) {
+            revert CallerIsNotReleaseAdmin();
+        }
+        if (royaltyAmount > 2_000) revert InvalidAmount();
+
+        numberOfReleases++;
+        _uris[numberOfReleases] = uri_;
+        /// Create splits address here
+        IReleaseRegistration(_catalog).registerRelease(trackIds, uri_, numberOfReleases);
+        _mint(receiver, numberOfReleases, amount, "");
+        _setTokenRoyalty(numberOfReleases, beneficiary, royaltyAmount);
+
+        emit ReleaseCreated(numberOfReleases);
+    }
+
+    /**
+     * @inheritdoc IReleases
+     */
+    function withdrawRelease(
+        address receiver,
+        uint256 tokenId,
+        uint256 amount
+    ) external onlyRole(RELEASE_ADMIN_ROLE) nonReentrant {
+        _requireTokenIdExists(tokenId);
+        _safeTransferFrom(address(this), receiver, tokenId, amount, "");
+        emit ReleaseWithdrawn(msg.sender, tokenId, amount);
+    }
+
+    /**
+     * @inheritdoc IReleases
+     */
+    function info() external view returns (uint256, bool) {
+        return (numberOfReleases, _isOpen);
+    }
+
+    /**
+     * @inheritdoc IReleases
+     */
+    function setUri(uint256 tokenId, string calldata uri_) external onlyRole(DEFAULT_ADMIN_ROLE) {
```

Comment:
How does this work if it is "Open"? Can the original creator update the URI?

If we need to start keeping track of who the token creator and have extra logic all over the place then I suggest we create a second contract for that use case